client/utils/account-fees.tsx

```diff
@@ -111,7 +128,7 @@ export const formatMethodFeesTooltip = (
 		<div className={ 'wcpay-fees-tooltip' }>
 			<div>
 				<div>Base fee</div>
-				<div>{ getFeeDescriptionString( currentBaseFee ) }</div>
+				<div>{ getFeeDescriptionString( tooltipAndPillBaseFee ) }</div>
```

Comment:
IMO this is the key to fixing this bug. I think you can agree with me that the problem we have is the generation of the fee description here, and also the 'Total per transaction' part. It shows blank because `getFeeDescriptionString` returns an empty string in the case where `percentage_rate` or `fixed_rate` is missing.

But I would go with a different approach. I'd like to introduce a new function. Just a wrapper around `getFeeDescriptionString`, something like:

```
// If the current base fee is missing a fixed or percentage rate, use the base fee's rate instead.
const getFeeDescriptionStringWithBaseFallback = (
	currentFee: BaseFee,
	baseFee: BaseFee
) => {
	if ( ! currentFee.percentage_rate ) {
		currentFee.percentage_rate = baseFee.percentage_rate;
	}

	if ( ! currentFee.fixed_rate ) {
		currentFee.fixed_rate = baseFee.fixed_rate;
	}

	return getFeeDescriptionString( currentFee );
};

```
So we would call this function in this line (and also to build the total):
```suggestion
				<div>{ getFeeDescriptionStringWithBaseFallback( currentBaseFee, accountFees.base ) }</div>
```
If you look carefully, this is just a different way of applying the fix you already have. IMO it is just shorter, and I also believe it helps to convey the intention and clarify things. 

What do you think?

Reply:
This is a beautiful and simplified alternative! I also like how it leaves the original function untouched, thereby avoiding the chance of any side-effects. Thanks so much, Miguel! I will rework this 👍

Reply:
It looks like the Base Fee message part on the tooltip  requires the ready formatted string generated by `getFeeDescriptionString` function.  However for Total per transaction display, there is a `total` variable that requires the fee itself (with fall back if the percentage and fixed fee is absent).

Hence, I have created a function -[ getCurrentFeeWithBaseFallBack ](https://github.com/Automattic/woocommerce-payments/blob/ae30b742c5a0880919f363cfabc64dbd6b4e4c1c/client/utils/account-fees.tsx#L89)

The base fee section of the tooltip, uses the output of the above function, and gets a formatted string via existing `getFeeDescriptionString`

The total fee uses the currentFee computed with fallback, adds the other fees. This computed total fee is then displayed using the same `getFeeDescriptionString` function.

Reply:
Marking this thread as resolved since the code has been changed.